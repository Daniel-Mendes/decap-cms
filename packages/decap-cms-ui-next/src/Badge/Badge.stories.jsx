import React from 'react';
import { action } from '@storybook/addon-actions';

import { Badge, BadgeGroup } from '.';
import color from '../utils/color';

export default {
  title: 'Components/Badge',
};

export function _Badge(args) {
  const { onClick, onDelete } = args;

  return (
    <Badge
      {...args}
      onClick={onClick ? action('onClick') : null}
      onDelete={onDelete ? action('onDelete') : null}
    />
  );
}

_Badge.argTypes = {
  children: { control: 'text' },
  color: {
    control: 'select',
    options: Object.keys(color).reduce((acc, color) => ({ ...acc, [color]: color }), {}),
  },
  variant: {
    control: 'select',
    options: ['solid', 'soft', 'outline'],
  },
  radius: { control: 'select', options: ['none', 'large', 'full'] },
  size: {
    control: 'select',
    options: ['sm', 'md', 'lg'],
  },
  hasMenu: { control: 'boolean' },

  onClick: { control: 'boolean' },
  onDelete: { control: 'boolean' },
};

_Badge.args = {
  children: 'Badge',
  color: 'neutral',
  variant: 'solid',
  radius: 'large',
  size: 'sm',
  hasMenu: false,

  onClick: false,
  onDelete: false,
};

export function _BadgeGroup(args) {
  const { size, onClick, hasMenu, onDelete } = args;

  return (
    <BadgeGroup {...args}>
      <Badge
        size={size}
        color="green"
        onClick={onClick ? action('onClick', event) : null}
        hasMenu={hasMenu}
        onDelete={onDelete ? action('onDelete', event) : null}
      >
        Apple
      </Badge>
      <Badge
        size={size}
        color="yellow"
        onClick={onClick ? () => alert('Badge clicked.') : null}
        hasMenu={hasMenu}
        onDelete={onDelete ? () => alert('Deleted tag') : null}
      >
        Banana
      </Badge>
      <Badge
        size={size}
        color="orange"
        onClick={onClick ? () => alert('Badge clicked.') : null}
        hasMenu={hasMenu}
        onDelete={onDelete ? () => alert('Deleted tag') : null}
      >
        Orange
      </Badge>
      <Badge
        size={size}
        color="red"
        onClick={onClick ? () => alert('Badge clicked.') : null}
        hasMenu={hasMenu}
        onDelete={onDelete ? () => alert('Deleted tag') : null}
      >
        Cherry
      </Badge>
      <Badge
        size={size}
        color="pink"
        onClick={onClick ? () => alert('Badge clicked.') : null}
        hasMenu={hasMenu}
        onDelete={onDelete ? () => alert('Deleted tag') : null}
      >
        Strawberry
      </Badge>
      <Badge
        size={size}
        color="purple"
        onClick={onClick ? () => alert('Badge clicked.') : null}
        hasMenu={hasMenu}
        onDelete={onDelete ? () => alert('Deleted tag') : null}
      >
        Grape
      </Badge>
      <Badge
        size={size}
        color="blue"
        onClick={onClick ? () => alert('Badge clicked.') : null}
        hasMenu={hasMenu}
        onDelete={onDelete ? () => alert('Deleted tag') : null}
      >
        Blueberry
      </Badge>
    </BadgeGroup>
  );
}

_BadgeGroup.argTypes = {
  direction: {
    control: 'select',
    options: ['horizontal', 'vertical'],
  },
};

_BadgeGroup.args = {
  direction: 'horizontal',
};
