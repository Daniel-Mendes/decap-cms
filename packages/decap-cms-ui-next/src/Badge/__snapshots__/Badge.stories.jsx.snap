// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Badge _Badge smoke-test 1`] = `"<div color="neutral" radius="large" class="css-6930ng-BadgeWrap-BadgeWrap epfx4ki3">Badge</div>"`;

exports[`Components/Badge _BadgeGroup smoke-test 1`] = `"<div direction="horizontal" class="css-1edslqn-BadgeGroup e154evrq0"><div color="green" radius="large" class="css-8hgyys-BadgeWrap-BadgeWrap epfx4ki3">Apple</div><div color="yellow" radius="large" class="css-h64jdo-BadgeWrap-BadgeWrap epfx4ki3">Banana</div><div color="orange" radius="large" class="css-2lm8dq-BadgeWrap-BadgeWrap epfx4ki3">Orange</div><div color="red" radius="large" class="css-11poh1o-BadgeWrap-BadgeWrap epfx4ki3">Cherry</div><div color="pink" radius="large" class="css-1txaaf-BadgeWrap-BadgeWrap epfx4ki3">Strawberry</div><div color="purple" radius="large" class="css-10l1sh1-BadgeWrap-BadgeWrap epfx4ki3">Grape</div><div color="blue" radius="large" class="css-1f2xm2u-BadgeWrap-BadgeWrap epfx4ki3">Blueberry</div></div>"`;
